sw/device/silicon_creator/rom_ext/rom_ext.c

```diff
@@ -541,47 +545,49 @@ static rom_error_t boot_svc_min_sec_ver_handler(boot_svc_msg_t *boot_svc_msg,
 }
 
 OT_WARN_UNUSED_RESULT
-static rom_error_t rom_ext_try_boot(void) {
-  boot_data_t boot_data;
-  HARDENED_RETURN_IF_ERROR(boot_data_read(lc_state, &boot_data));
-
-  boot_svc_msg_t boot_svc_msg = retention_sram_get()->creator.boot_svc_msg;
-  if (boot_svc_msg.header.identifier == kBootSvcIdentifier) {
-    HARDENED_RETURN_IF_ERROR(boot_svc_header_check(&boot_svc_msg.header));
-    uint32_t msg_type = boot_svc_msg.header.type;
+static rom_error_t handle_boot_svc(boot_data_t *boot_data) {
+  boot_svc_msg_t *boot_svc_msg = &retention_sram_get()->creator.boot_svc_msg;
+  // TODO(lowRISC#22387): Examine the boot_svc code paths for boot loops.
+  if (boot_svc_msg->header.identifier == kBootSvcIdentifier) {
+    HARDENED_RETURN_IF_ERROR(boot_svc_header_check(&boot_svc_msg->header));
+    uint32_t msg_type = boot_svc_msg->header.type;
     switch (launder32(msg_type)) {
       case kBootSvcEmptyType:
         HARDENED_CHECK_EQ(msg_type, kBootSvcEmptyType);
         break;
       case kBootSvcNextBl0SlotReqType:
         HARDENED_CHECK_EQ(msg_type, kBootSvcNextBl0SlotReqType);
-        HARDENED_RETURN_IF_ERROR(
-            boot_svc_next_boot_bl0_slot_handler(&boot_svc_msg, &boot_data));
-        break;
+        return boot_svc_next_boot_bl0_slot_handler(boot_svc_msg, boot_data);
       case kBootSvcPrimaryBl0SlotReqType:
         HARDENED_CHECK_EQ(msg_type, kBootSvcPrimaryBl0SlotReqType);
-        HARDENED_RETURN_IF_ERROR(
-            boot_svc_primary_boot_bl0_slot_handler(&boot_svc_msg, &boot_data));
-        break;
+        return boot_svc_primary_boot_bl0_slot_handler(boot_svc_msg, boot_data);
       case kBootSvcMinBl0SecVerReqType:
         HARDENED_CHECK_EQ(msg_type, kBootSvcMinBl0SecVerReqType);
-        HARDENED_RETURN_IF_ERROR(
-            boot_svc_min_sec_ver_handler(&boot_svc_msg, &boot_data));
-        break;
+        return boot_svc_min_sec_ver_handler(boot_svc_msg, boot_data);
+      case kBootSvcOwnershipUnlockReqType:
+        HARDENED_CHECK_EQ(msg_type, kBootSvcOwnershipUnlockReqType);
+        return ownership_unlock_handler(boot_svc_msg, boot_data);
       case kBootSvcNextBl0SlotResType:
+      case kBootSvcPrimaryBl0SlotResType:
+      case kBootSvcMinBl0SecVerResType:
+      case kBootSvcOwnershipUnlockResType:
         // For response messages left in ret-ram we do nothing.
         break;
       default:
-        HARDENED_TRAP();
+        dbg_printf("Unknown boot_svc: %x\r\n", msg_type);
```

Comment:
1) should we return an error here? Or are we confident that an attacker will not be able to leverage this fall-through condition.

2) An attacker may be able to use this print statement to characterize faults to the SRAM holding the boot service command (e.g. [fault parameters] == [observed %x behavior]). The gained information then could then later be used to mount an attack at runtime. @johannheyszl, is this something we should be concerned about, or do you think the SRAM re-scrambling will help to mitigate concerns here.

A less attacker friendly message would be `"Unknown boot_srv message type"` at the cost of usability.

Reply:
Thx @moidx - I think you are right it would be better not output the value.

AFAIK, the retention SRAM is scrambled using a default, global netlist key (#7895 ), so there is no rescrambling that would help.

Reply:
Agreed: we shouldn't print anything for this case.  Removed.